plugins/chain_plugin/include/eosio/chain_plugin/blockvault_sync_strategy.hpp

```diff
@@ -9,30 +9,49 @@ namespace eosio {
         struct blockvault_sync_strategy : public sync_callback {
             blockvault_sync_strategy(block_vault_interface* blockvault, BP& blockchain_provider, std::function<void()> shutdown,
                                      std::function<bool()> check_shutdown) :
-                _got_snapshot(false), _blockvault(blockvault), _blockchain_provider(blockchain_provider), _shutdown(shutdown), _check_shutdown(check_shutdown) {
+                _startup_run(false), _blockvault(blockvault), _blockchain_provider(blockchain_provider), _shutdown(shutdown), _check_shutdown(check_shutdown) {
+            }
+
+            void run_startup() {
+                if (_blockchain_provider.genesis) {
+                    _blockchain_provider.chain->startup(_shutdown, _check_shutdown, *_blockchain_provider.genesis);
+                }else {
+                    _blockchain_provider.chain->startup(_shutdown, _check_shutdown);
+                }
+                _startup_run = true;
+
             }
 
             void do_sync() {
                 auto head_block = _blockchain_provider.chain->fork_db().head();
```

Comment:
Seems odd to me that this would reach all the way into fork_db and grab the head.

Reply:
Where should I be getting it from?

Reply:
Seems there should be an interface on chain_plugin to get the current known id (to be determined according to the question below).